### PR TITLE
StreamProtocols are passed to Identify

### PR DESCRIPTION
**Pull request by ErakhtinB**

### CODE REVIEW

File: include/libp2p/peer/stream_protocols.hpp [hunk 1]
```diff
@@ -16,5 +16,11 @@ namespace libp2p {
    * Used by `Host::newStream`, `Dialer::newStream`, `Host::setProtocolHandler`,
    * `Router::setProtocolHandler`.
    */
-  using StreamProtocols = std::vector<peer::ProtocolName>;
+  struct StreamProtocols {
```
Comment (turuslan) [inline include/libp2p/peer/stream_protocols.hpp:19]: Revert.
`StreamProtocols` is just data type (like `vector`),
it's ambigous to inject it.

Create `struct Config` for identify (see ping or kademlia for example)

**Review by turuslan — CHANGES_REQUESTED** [review]

File: include/libp2p/peer/stream_protocols.hpp (hunk 1 above)
Comment (xDimon) [reply include/libp2p/peer/stream_protocols.hpp:19]: Might to use Tagged wrapping-class from kagome

File: include/libp2p/protocol/identify/config.hpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <libp2p/peer/stream_protocols.hpp>
+
+namespace {
```
Comment (turuslan) [inline include/libp2p/protocol/identify/config.hpp:11]:
```suggestion
namespace libp2p::protocol {
```

File: test/libp2p/protocol/identify_test.cpp [hunk 3]
```diff
@@ -80,7 +80,9 @@ class IdentifyTest : public testing::Test {
 
     id_msg_processor_ = std::make_shared<IdentifyMessageProcessor>(
         host_, conn_manager_, id_manager_, key_marshaller_);
-    identify_ = std::make_shared<Identify>(host_, id_msg_processor_, bus_);
+    IdentifyConfig config;
+    config.protocols = {kIdentifyProto};
```
Comment (turuslan) [inline test/libp2p/protocol/identify_test.cpp:84]: May use default
```suggestion
```

**Review by turuslan — APPROVED** [review]

File: include/libp2p/protocol/identify/config.hpp [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  const std::string kIdentifyProto = "/ipfs/id/1.0.0";
+}  // namespace
+
+namespace libp2p::protocol {
```
Comment (igor-egorov) [inline include/libp2p/protocol/identify/config.hpp:15]:
```suggestion

```
The change completes the previous

**Review by igor-egorov — APPROVED** [review]

**Pull request merged**